### PR TITLE
Api enhancements

### PR DESCRIPTION
**Pull request by drahnieR**
PR #1308 breaks so start a new one

~~Distinguish entries in flow:conn:* zset by destination port~~ (solved in #1308)
add active alarm count to init firewalla/firewalla.ios#1019
implement count, ts, asc options for get alarms request firewalla/firewalla.ios#1020

### CODE REVIEW

**Comment by drahnieR** [comment]
Fixed some bugs in policy

File: net2/NetBotTool.js [hunk 1]
```diff
@@ -310,17 +310,16 @@ class NetBotTool {
...
-      })
+    traffic.forEach((f) => {
+      f.begin = begin;
+      f.end = end;
+    })
 
-      let promises = Promise.all(traffic.map((f) => {
-        return intelTool.getIntel(f.ip)
-        .then((intel) => {
+    traffic = await Promise.all(traffic.map((f) => {
```
Comment (MelvinTo) [inline net2/NetBotTool.js]: can be optimized with native async/await here

File: alarm/PolicyManager2.js [hunk 2]
```diff
@@ -314,34 +314,38 @@ class PolicyManager2 {
...
-    } else {
-      return Promise.reject(new Error("UpdatePolicyAsync requires policy ID"))
+    let existing = await this.getPolicy(policy.pid);
+
+    Object.assign(existing, policy);
+
+    await rclient.hmsetAsync(policyKey, existing.redisfy());
+
+    if (policy.expire === '') {
+      await (rclient.hdelAsync(policyKey, "expire"))
```
Comment (MelvinTo) [inline alarm/PolicyManager2.js]: remove the bracket () for awaits

File: net2/FlowTool.js [hunk 3]
```diff
@@ -267,28 +267,17 @@ class FlowTool {
       f.device = flow.mac;
     }
 
-    if(flow.pf) {
+    f.ptocotol = flow.pr;
```
Comment (MelvinTo) [inline net2/FlowTool.js]: typo here

File: net2/FlowTool.js [hunk 4]
```diff
@@ -267,28 +267,17 @@ class FlowTool {
...
+
+    try {
       if(flow.lh === flow.sh) {
-        try {
-          const protocol = Object.keys(flow.pf)[0].split(".")[0];
```
Comment (MelvinTo) [inline net2/FlowTool.js]: the code logic is deleted??

File: net2/DNSManager.js [hunk 5]
```diff
@@ -49,98 +49,93 @@ const DNSQUERYBATCHSIZE=5;
...
+      } else {
+        throw new Error('IP Not Found: ' + ip);
+      }
+    } else {
+      log.error("ResolveHost:BadIP", ip);
+      throw new Error('bad ip');
     }
 
-/*
+    return rclient.hgetallAsync("host:mac:" + mac)
```
Comment (MelvinTo) [inline net2/DNSManager.js]: could reuse the functions from HostTool.js

e.g. https://github.com/firewalla/firewalla/blob/master/net2/HostTool.js#L72

File: net2/FlowTool.js (hunk 4 above)
Comment (drahnieR) [reply net2/FlowTool.js]: https://github.com/firewalla/firewalla/blob/38f012f9cec31abaee04075435a18dd494e69c83/net2/BroDetect.js#L883-L884
no need to do string parsing here, flow.pr, flow.sp, flow.dp have all things we need

File: net2/NetBotTool.js (hunk 1 above)
Comment (MelvinTo) [reply net2/NetBotTool.js]: fake example:

```
await Promise.all(traffic.map( **async** (f) => {
        const intel = await intelTool.getIntel(f.ip);
        if(intel) {
            f.country = intel.country;
            f.host = intel.host
```

**Comment by drahnieR** [comment]
updated, more code clean up.

**Pull request merged**